core/src/main/java/io/jenkins/pluginhealth/scoring/probes/IncrementalBuildDetectionProbe.java

```diff
@@ -0,0 +1,147 @@
+
+        if (mavenExtensionsFile.isPresent() && mavenConfigFile.isPresent()) {
+            return isExtensionsXMLConfigured(mavenExtensionsFile.get()) && isMavenConfigConfigured(mavenConfigFile.get())
+                ? ProbeResult.success(key(), String.format("Incremental Build is configured in the %s plugin.", plugin.getName()))
+                : ProbeResult.failure(key(), String.format("Incremental Build is not configured in the %s plugin.", plugin.getName()));
+        }
+        return ProbeResult.failure(key(), String.format("Incremental Build is not configured in the %s plugin.", plugin.getName()));
+    }
+
+    @Override
+    public String[] getProbeResultRequirement() {
+        return new String[]{ContinuousDeliveryProbe.KEY};
+    }
+
+    @Override
+    public String key() {
+        return KEY;
+    }
+
+    @Override
+    public String getDescription() {
+        return "The probe detects whether incremental build is configured in Maven Config.";
+    }
+
+    /**
+     * Checks whether `extensions.xml` is configured in the plugin
+     *
+     * @param path Looks for extensions.xml configuration in the particular path
+     * @return true if a correct configuration is found, otherwise false
+     */
+    private boolean isExtensionsXMLConfigured(Path path) {
+        try {
+            DocumentBuilderFactory dbFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder dBuilder = dbFactory.newDocumentBuilder();
+            Document doc = dBuilder.parse(path.toFile());
+            doc.getDocumentElement().normalize();
+            Element rootElement = doc.getDocumentElement();
+            Element extensionElement = (Element) rootElement.getElementsByTagName("extension").item(0);
+            Element groupIdElement = (Element) extensionElement.getElementsByTagName("groupId").item(0);
+            Element artifactIdElement = (Element) extensionElement.getElementsByTagName("artifactId").item(0);
+            String groupId = groupIdElement.getTextContent();
+            String artifactId = artifactIdElement.getTextContent();
+            return groupId.equals(INCREMENTAL_TOOL) && artifactId.equals(INCREMENTAL_TOOL_ARTIFACT_ID);
+        } catch (IOException e) {
+            LOGGER.error("Could not read the file during probe {}. {}", key(), e);
+        } catch (ParserConfigurationException | SAXException e) {
+            LOGGER.error("Could not parse the file during probe {}. {}", key(), e);
+        }
+        return false;
+    }
+
+    /**
+     * Checks whether `maven.config` is configured in the plugin
+     *
+     * @param path Looks for extensions.xml configuration in the particular path
+     * @return true if a correct configuration is found, otherwise false
+     */
+    private boolean isMavenConfigConfigured(Path path) {
+        try {
+            return Files.lines(path).toList().containsAll(List.of("-Pconsume-incrementals", "-Pmight-produce-incrementals"));
```

Comment:
I think instead of `lines().toList()` you can directly use `.readAllLines()` which returns a `List` and not a `Stream`.